dataplane/kernel_interface_handle.cpp

```diff
@@ -0,0 +1,157 @@
+#include <linux/if.h>
+#include <sys/ioctl.h>
+#include <sys/un.h>
+
+#include <rte_ethdev.h>
+
+#include "kernel_interface_handle.h"
+namespace dataplane
+{
+
+bool KernelInterfaceHandle::SetUp() const
+{
+	int socket = ::socket(AF_INET, SOCK_DGRAM, 0);
+	if (socket < 0)
+	{
+		return false;
+	}
+
+	struct ifreq request;
+	memset(&request, 0, sizeof request);
+
+	strncpy(request.ifr_name, m_vdev_name.data(), IFNAMSIZ);
+
+	request.ifr_flags |= IFF_UP;
+	if (auto res = ioctl(socket, SIOCSIFFLAGS, &request))
+	{
+		YANET_LOG_ERROR("failed to set interface %s up, ioctl returned (%d)", m_vdev_name.data(), res);
+		return false;
+	}
+	return true;
+}
+
+KernelInterfaceHandle::~KernelInterfaceHandle()
+{
+	if (Valid())
+		Remove();
+}
+KernelInterfaceHandle::KernelInterfaceHandle(KernelInterfaceHandle&& other)
```

Comment:
Separate definitions with an empty line, please.
Sometimes it is done, sometimes not, as seen here